classes/author/creditRole/Repository.php

```diff
@@ -0,0 +1,121 @@
+        $expiration = \DateInterval::createFromDateString(static::MAX_CACHE_LIFETIME);
+
+        $setRoles = fn (array $roles, string $localeKey): array => [
+            'roles' => $roles,
+            'degrees' => [
+                CreditRoleDegree::NULL->getLabel() => '',
+                CreditRoleDegree::LEAD->getLabel() => __('submission.submit.creditRoles.degrees.lead', [], $localeKey),
+                CreditRoleDegree::EQUAL->getLabel() => __('submission.submit.creditRoles.degrees.equal', [], $localeKey),
+                CreditRoleDegree::SUPPORTING->getLabel() => __('submission.submit.creditRoles.degrees.supporting', [], $localeKey),
+            ],
+        ];
+        $getJson = fn (string $file): array => is_array($json = json_decode(file_get_contents($file) ?: "", true)) ? $json : [];
+        $getRoles = fn (array $json): array => Arr::map($json['translations'] ?? [], fn (array $items) => $items['name']);
+
+        // Init with ui locales and default locale
+        if (!$this->creditRoleTerms) {
+            $this->creditRoleTerms = Cache::remember($key, $expiration, fn () =>
+                collect(Application::get()->getRequest()->getContext()->getSupportedLocales())
+                    ->push($this->defaultLocale)
+                    ->unique()
+                    ->mapWithKeys(fn (string $l): array => [$l => $this->getFileName($l)])
+                    ->map(fn (?string $filename, string $l): ?array => $filename ? $setRoles($getRoles($getJson($filename)), $l) : null)
+                    ->toArray()
+            );
+        }
+
+        // Add new locale if not in array
+        if (!isset($this->creditRoleTerms[$locale])) {
+            $filename = $this->getFileName($locale);
+            $this->creditRoleTerms[$locale] = $filename ? $setRoles($getRoles($getJson($filename)), $locale) : null;
+            Cache::put($key, $this->creditRoleTerms, $expiration);
+        }
+    }
+
+    /**
+     * Get file name of the localized terms
+     */
+    protected function getFileName(string $locale): ?string
+    {
+        $prefix = Core::getBaseDir() . '/' . PKP_LIB_PATH . "/lib/creditRoles/translations/";
+        $suffix = '.json';
+        // Determine the language and region we're looking for from $locale
+        $language = \Locale::getPrimaryLanguage($locale);
+        $region = \Locale::getRegion($locale) ?? null;
+        $localeAndRegion = $language . ($region ? "_{$region}" : '');
+        // Get a list of available options from the filesystem and return file name
+        return collect(glob("{$prefix}*{$suffix}"))
+            ->first(fn (string $filename) =>
+                // 1. Look for an exact match and return it.
+                ($filename === "{$prefix}{$locale}{$suffix}") ||
+                // 2. Look in the preference list for a preferred fallback.
+                (isset($this->localeMapping[$localeAndRegion]) && $filename === "{$prefix}{$this->localeMapping[$localeAndRegion]}{$suffix}") ||
```

Comment:
this can be $this->localeMapping[$locale] -- because we map our locales to those of credit